src/main/kotlin/org/jetbrains/research/testspark/tools/TestProcessor.kt

```diff
@@ -100,15 +100,22 @@ class TestProcessor(
             } else {
                 "-javaagent:$jacocoAgentLibraryPath=destfile=$dataFileName.exec,append=false"
             }
+
+        // save classpaths to a temp file
+        val classPathsList =
+            testCompiler.getClassPaths(
+                projectBuildPath,
+            ) + "${DataFilesUtil.classpathSeparator}${junitRunnerLibraryPath}${DataFilesUtil.classpathSeparator}$resultPath"
+        val fileName = DataFilesUtil.makeTmpFile("testSparkCP", classPathsList)
+        val classPaths = "@$fileName"
+
         val testExecutionResult =
             CommandLineRunner.run(
                 arrayListOf(
-                    javaRunner.absolutePath,
+                    "\"${javaRunner.absolutePath}\"",
                     javaAgentFlag,
                     "-cp",
-                    "\"${testCompiler.getClassPaths(
```

Comment:
I've noticed that previously we were surrounding the class paths with quotation marks, presumably to handle cases where we have whitespace. Do you know if @filename command takes care of it by default, or it also needs to be surrounded with quotation marks?

Reply:
It definitively needs to be surrounded with quotation marks. Good catch!